woocommerce/woocommerce-android
# Shipping Labels: fix international DHL purchase

Closes: #4735 

### Description
This PR includes two changes:
1. It uses the change from the fluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2111 to pass is_letter value when purchasing a shipping label.
2. It changes the rules to require a destination phone number for all international shipping labels, this aligns Android with the web client and iOS.

At first, we thought that the missing `is_letter` field is the cause of international DHL failures, but @itsmeichigo found the real cause (ref p1631189289017100/1629875137.007200-slack-C026HH75ECA), which is fetching the rates without a destination phone number, then adding a phone number for the purchase, this happened only in Android because we didn't require the destination phone number until the user chooses DHL as a carrier, and this should be fixed by the change 2.

Please don't merge before merging the FluxC's PR

### Testing instructions
1. Create an international order, and make sure your store is eligible for shipping label creation.
2. Open the order in the app.
3. Click on Create shipping label.
4. Confirm that the destination address requires you to enter a phone number.
5. Select DHL as a carrier in the rates screen.
6. Confirm that the purchase succeeds.
